src/Umbraco.Infrastructure/PropertyEditors/RichTextEditorPastedImages.cs

```diff
@@ -56,149 +67,301 @@ public sealed class RichTextEditorPastedImages
+    }
+
+    public RichTextEditorPastedImages(
+        IUmbracoContextAccessor umbracoContextAccessor,
+        ILogger<RichTextEditorPastedImages> logger,
+        IHostingEnvironment hostingEnvironment,
+        IMediaService mediaService,
+        IContentTypeBaseServiceProvider contentTypeBaseServiceProvider,
+        MediaFileManager mediaFileManager,
+        MediaUrlGeneratorCollection mediaUrlGenerators,
+        IShortStringHelper shortStringHelper,
+        IPublishedUrlProvider publishedUrlProvider,
+        IImageUrlGenerator imageUrlGenerator,
+        IOptions<ContentSettings> contentSettings)
+    {
+        _umbracoContextAccessor =
+            umbracoContextAccessor ?? throw new ArgumentNullException(nameof(umbracoContextAccessor));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _hostingEnvironment = hostingEnvironment;
+        _mediaService = mediaService ?? throw new ArgumentNullException(nameof(mediaService));
+        _contentTypeBaseServiceProvider = contentTypeBaseServiceProvider ??
+                                          throw new ArgumentNullException(nameof(contentTypeBaseServiceProvider));
+        _mediaFileManager = mediaFileManager;
+        _mediaUrlGenerators = mediaUrlGenerators;
+        _shortStringHelper = shortStringHelper;
+        _publishedUrlProvider = publishedUrlProvider;
+        _imageUrlGenerator = imageUrlGenerator;
+        _contentSettings = contentSettings.Value;
+
+        _tempFolderAbsolutePath = _hostingEnvironment.MapPathContentRoot(Constants.SystemDirectories.TempImageUploads);
     }
 
     /// <summary>
-    ///     Used by the RTE (and grid RTE) for drag/drop/persisting images
+    ///     Used by the RTE (and grid RTE) for converting inline base64 images to Media items.
     /// </summary>
-    public string FindAndPersistPastedTempImages(string html, Guid mediaParentFolder, int userId, IImageUrlGenerator imageUrlGenerator)
+    /// <param name="html">HTML from the Rich Text Editor property editor.</param>
+    /// <param name="mediaParentFolder"></param>
+    /// <param name="userId"></param>
+    /// <returns>Formatted HTML.</returns>
+    /// <exception cref="NotSupportedException">Thrown if image extension is not allowed</exception>
+    internal string FindAndPersistEmbeddedImages(string html, Guid mediaParentFolder, int userId)
     {
         // Find all img's that has data-tmpimg attribute
         // Use HTML Agility Pack - https://html-agility-pack.net
         var htmlDoc = new HtmlDocument();
         htmlDoc.LoadHtml(html);
 
-        HtmlNodeCollection? tmpImages = htmlDoc.DocumentNode.SelectNodes($"//img[@{TemporaryImageDataAttribute}]");
-        if (tmpImages == null || tmpImages.Count == 0)
+        HtmlNodeCollection? imagesWithDataUris = htmlDoc.DocumentNode.SelectNodes("//img");
+        if (imagesWithDataUris == null || imagesWithDataUris.Count == 0)
```

Comment:
Lets change this to pattern matching while we're at it 🔥 
```suggestion
        if (imagesWithDataUris is null || imagesWithDataUris.Count is 0)
```

Reply:
You can even use [property pattern matching](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#property-pattern) to make this more concise 😄
```suggestion
        if (imagesWithDataUris is null or { Count: 0 })
```